lib/screens/home/view_more/hidden_proposal_creation/hidden_proposal_creation_screen.dart

```diff
@@ -0,0 +1,276 @@
+              minimumSize: Size(
+                80,
+                50,
+              ),
+              onPressed: () {
+                final data = _bloc.getMessageJson();
+                Navigator.of(context).push(PwDataScreen(
+                  title: strings.stakingConfirmData,
+                  data: data,
+                ).route());
+              },
+              child: PwText(
+                strings.stakingConfirmData,
+                style: PwTextStyle.footnote,
+                underline: true,
+              ),
+            ),
+          ),
+        ],
+      ),
+      body: Container(
+        color: Theme.of(context).colorScheme.neutral750,
+        child: StreamBuilder<ProposalCreationDetails?>(
+            initialData: _bloc.creationDetails.value,
+            stream: _bloc.creationDetails,
+            builder: (context, snapshot) {
+              final details = snapshot.data;
+
+              if (details == null || details.isLoading) {
+                return Center(
+                  child: CircularProgressIndicator(),
+                );
+              }
+
+              return Column(
+                children: [
+                  Expanded(
+                    child: ListView(
+                      padding: EdgeInsets.symmetric(horizontal: Spacing.large),
+                      children: [
+                        DetailsHeader(
+                          title: strings.devCreateProposalDetails,
+                        ),
+                        PwListDivider.alternate(),
+                        StakingTextFormField(
+                          hint: strings.devProposalTitle,
+                          textEditingController: _titleEditingController,
+                          keyboardType: TextInputType.name,
+                        ),
+                        VerticalSpacer.large(),
+                        PwListDivider.alternate(),
+                        VerticalSpacer.large(),
+                        StakingTextFormField(
+                          hint: strings.devProposalDescription,
+                          textEditingController: _descriptionEditingController,
+                          keyboardType: TextInputType.name,
+                        ),
+                        VerticalSpacer.large(),
+                        PwListDivider.alternate(),
+                        DetailsItem.fromStrings(
+                          title: strings.stakingDelegateAvailableBalance,
+                          value:
+                              strings.hashAmount(details.hashAmount.toString()),
+                        ),
+                        PwListDivider.alternate(),
+                        DetailsItem.withRowChildren(
+                          title: strings.devInitialDepositAmount,
+                          children: [
+                            PwText(
+                              strings.hashAmount(
+                                  details.initialDeposit.toInt().toString()),
```

Comment:
Thinking that truncating with `toInt` might not be desirable.

Reply:
I was getting rounding errors and it looked awful. Maybe going the `Decimal` route will make it not as bad.